### PR TITLE
Deploy pipeline

### PR DESCRIPTION
**Pull request by kamaldodiya**
The PR Deploys metadata-api to New k8s cluster in the ensembl-prod namespace. 
Later we should add more jobs to deploy to other environment

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,2 @@
-PORT=8083
+PORT=8014
 DEBUG=False
```
Comment (tuanebi) [inline .env:2]: Are this DEBUG and DEBUG_MODE in configmap the same thing?

File: .env [hunk 2]
```diff
@@ -1,2 +1,2 @@
-PORT=8083
+PORT=8014
```
Comment (tuanebi) [inline .env:1]: PORT is set here but is hardcoded in Dockerfile and yaml files. Can it be improved?

Comment (kamaldodiya) [reply .env:1]: I think @Mehrnaz-Charkhchi  can comment on this. But my interpretation is .env is used for local development if one  wants to run it at different port then this is the place where it should be modified.

File: .env (hunk 1 above)
Comment (kamaldodiya) [reply .env:2]: Updated configmap to be consistent with this.

File: .env (hunk 2 above)
Comment (azangru) [reply .env:1]: Wouldn't you normally commit an `.env.sample`, from which the developer then creates a git-ignored `.env` file with appropriate modifications for local development? Such as is done in the [genome browser repository](https://github.com/Ensembl/ensembl-dauphin-style-compiler/tree/master/configurations/dev)?

_(Just curious. I don't have a strong opinion on the matter.)_

File: k8s/base/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      containers:
+        - name: metadata-api
+          image: DOCKER_IMAGE
+          ports:
+            - containerPort: 8014
+          imagePullPolicy: Always
+          envFrom:
+            - configMapRef:
+                name: metadata-api-configmap
+          command: ["uvicorn"]
```
[the remark this answers is not in the document]
Comment (tuanebi) [reply k8s/base/deployment.yaml:26]: The docker document says do not use this image for K8s. Instead it suggests to run 1 uvicorn process and use K8s to handle replication. See the Warning at https://hub.docker.com/r/tiangolo/uvicorn-gunicorn

[the remark this answers is not in the document]
Comment (Mehrnaz-Charkhchi) [reply k8s/base/deployment.yaml:26]: @kamaldodiya & @tuanebi  In this case it seems like we need to update this repo's docker file and any other repo that's using this image.

[the remark this answers is not in the document]
Comment (kamaldodiya) [reply k8s/base/deployment.yaml:26]: Added Customs build Dockerfile

File: pyproject.toml [hunk 4]
```diff
@@ -13,12 +13,16 @@ starlette = "^0.13.4"
...
+<<<<<<< HEAD
 aiohttp = "^3.8.4"
 grpcio = "1.38.1"
 grpcio-tools = "1.38.1"
 uvicorn = "^0.22.0"
 orjson = "^3.9.1"
 black = "^23.3.0"
+=======
+uvicorn = "^0.22.0"
+>>>>>>> f21bef3 (Update dependencies - uvicorn,aiohttp)
```
Comment (azangru) [inline pyproject.toml:25]: A conflict ☝️

**Review by Mehrnaz-Charkhchi — APPROVED** [review]

**Review by tuanebi — APPROVED** [review]

**Pull request merged**